### PR TITLE
Skip create api server

### PR DESCRIPTION
**Pull request by sivchari**
#### What type of PR is this?

/assign @sivchari 

#### What this PR does / why we need it:
/area simulator

kube-scheduler-simulator absolutely create a Kubernetes API Server even if KubeAPIServerURL of configuration value is not empty. So this changes make kube-scheduler-simulator skip to create new Kubernetes API Server if KubeAPIServerURL is not empty.

#### Which issue(s) this PR fixes:

Related with #251 

#### Special notes for your reviewer:

/label tide/merge-method-squash

### CODE REVIEW

File: simulator/simulator.go [hunk 1]
```diff
@@ -35,11 +37,23 @@ func startSimulator() error {
 		return xerrors.Errorf("get config: %w", err)
 	}
 
-	restclientCfg, apiShutdown, err := k8sapiserver.StartAPIServer(cfg.KubeAPIServerURL, cfg.EtcdURL, cfg.CorsAllowedOriginList)
-	if err != nil {
-		return xerrors.Errorf("start API server: %w", err)
+	kubeAPIServerURL := cfg.KubeAPIServerURL
```
Comment (sanposhiho) [inline simulator/simulator.go:40]: KubeAPIServerURL has the default value. 
So, you gotta change something if you'd like to reuse this.

https://github.com/kubernetes-sigs/kube-scheduler-simulator/blob/0be3194647ec86ec8e0b3befb94c015ddf366c0c/simulator/config/config.go#L139-L156

Or, probably just adding `EnableKubeAPIServer` parameter to the config is the simplest way?

**Review by sanposhiho — COMMENTED** [review]
/ok-to-test

File: simulator/simulator.go (hunk 1 above)
Comment (sivchari) [reply simulator/simulator.go:40]: > Or, probably just adding EnableKubeAPIServer parameter to the config is the simplest way?

It looks good to me. How about like this ? We remove KubeAPIServer URL default value or remove KubeAPIServerURL field , add a new receiver method to return the string of KubeAPIServerURL. I think it's simple way. What do you think ?

Comment (sanposhiho) [reply simulator/simulator.go:40]: I don't quite get what you mean. You mean creating another env named `KubeAPIServerURL` and remove default values from `KUBE_API_PORT` and `KUBE_API_HOST`? (If wrong, please elaborate.

Comment (sivchari) [reply simulator/simulator.go:40]: Sorry, I don't know what I was getting at, too. Please forget it once.
I think it's good way to add new env named ExternalKubeAPIServerURL. If is's set, a simulator uses external kube-api-server. If not, a simulator uses internal kube-api-server which creates itself. What do you think ?

Comment (sanposhiho) [reply simulator/simulator.go:40]: sg

**Comment by sanposhiho** [comment]
/hold

We have to discuss fully the idea of integrating kwok without installing CLI in the issue. ([ref](https://github.com/kubernetes-sigs/kube-scheduler-simulator/issues/251#issuecomment-1837124903))
If we can, this PR wouldn't be needed anymore in the first place.

**Comment by sanposhiho** [comment]
/close

It's no longer valid and should be replaced with the PR to create kube-apiserver via [kubernetes-sigs/e2e-framework/support/kwok](https://github.com/kubernetes-sigs/e2e-framework/blob/c61338a3773c2b64550d0e46037bfe541fa54645/support/kwok/kwok.go#L135).

**Pull request closed without merging**